docs/DeveloperGuide.md

```diff
@@ -1187,28 +1196,92 @@ The `MathUtil` class contains generic functions for mathematical operations such
 
 ---
 
-## **Unit Tests**
+## **Testing Types**
+
+Types of tests used:
+* Developer Testing
+* Regression Testing
+* Unit Tests
+* Stubs
+* Integration testing
+* Dogfooding
+
+Our organisation made use of different testing types in order to cover multiple grounds. Types of testings also changed 
+with the progress of the product.
+
+### **Developer Testing**
+
+Our organisation opted to use developer testing right from the start with every increment of the product, instead of 
+waiting for the whole product to be ready before conducting more tests.
+
+This type of testing involved the developers trying out the product with the various commands implemented, to 
+discover possible bugs early, and avoid major reworks in future increments and iterations.
+
+### **Regression Testing**
+
+Our organisation continued with regression testing as increments came in, to ensure that previous features were not 
+broken with changes and modifications. This is due to the interconnectedness of certain features implemented, such as 
+commands and the model. We used a breadth-first iterative model to extensive test the wide scope of the project.
```

Comment:
I think you misunderstood what I say, the breadth-first iterative model is not a testing model, it is a SDLC model. You don't use that to "extensive test" the wide scope of the project. 

I would need you to fix the English as well.